### PR TITLE
UI Base package

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/ui/src/Button.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+				<Button after={<Search16 />} {...props}>
+					After
+				</Button>
+				<Button before={<CheckFill16 />} after={<Search16 />} {...props}>
+					Before & After
+				</Button>
+			</div>
+		);
+	},
+} satisfies Meta<typeof Button>;
```
Comment (williamrobertson13) [inline apps/ui/src/Button.stories.tsx:37]: nioce improved type safety 😉

File: apps/ui/src/Button.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	},
+};
+
+export const Secondary: Story = {
+	args: {
+		variant: 'secondary',
+	},
+};
+
+export const Outline: Story = {
```
Comment (williamrobertson13) [inline apps/ui/src/Button.stories.tsx:54]: A story using the `asChild` prop, i.e., `"As a link"`, would be nice

File: apps/ui/src/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		ButtonHTMLAttributes<HTMLButtonElement> {
+	loading?: boolean;
+	before?: ReactNode;
+	after?: ReactNode;
+	asChild?: boolean;
+}
+
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+	({ variant, size, loading, children, before, after, asChild, ...props }, ref) => {
+		const Comp = asChild ? Slot : 'button';
```
Comment (williamrobertson13) [inline apps/ui/src/Button.tsx:40]: If `asChild` is true, `Comp` will always be a `div` which seems not quite right

**Review by williamrobertson13 — APPROVED** [review]
lgtm besides one small thing - excited about this!

File: apps/ui/src/Button.tsx (hunk 3 above)
Comment (Jordan-Mysten) [reply apps/ui/src/Button.tsx:40]: Not sure what you mean by this?

Comment (williamrobertson13) [reply apps/ui/src/Button.tsx:40]: Updated my comment! Ignore me haha

File: apps/ui/src/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+		if (asChild && (loading || before || after)) {
```
Comment (williamrobertson13) [inline apps/ui/src/Button.tsx:44]: Is there any benefit to enforcing this at the type level or too overkill maybe? This works fine though haha

**Review by williamrobertson13 — APPROVED** [review]
Re-stamp!

**Review by Nikhil-Mysten — APPROVED** [review]

File: apps/ui/src/Button.tsx (hunk 4 above)
Comment (Jordan-Mysten) [reply apps/ui/src/Button.tsx:44]: It's certainly possible, I was just thinking it's probably overkill, and I'm slowly getting convinced that consumers should just do the before / after themselves (loading is still iffy, but there's probably a clever way to do it without needing to compose the children layout).

**Pull request merged**